drivers/spi/spi_gecko_eusart.c

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright (c) 2024 Daikin Comfort Technologies North America, Inc.
+ * Copyright (c) 2019 Christian Taedcke <hacking@taedcke.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT silabs_gecko_spi_eusart
+
+#define LOG_LEVEL CONFIG_SPI_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(spi_gecko_eusart);
+#include "spi_context.h"
+
+#include <zephyr/sys/sys_io.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/spi.h>
+#include <soc.h>
+
+#include "em_cmu.h"
+#include "em_eusart.h"
+
+#include <stdbool.h>
+#include <zephyr/drivers/pinctrl.h>
+
+#if (EUSART_COUNT == 1)
+#define CLOCK_EUSART(ref)
+	(((ref) == EUSART0) ? cmuClock_EUSART0 : -1)
+#elif (EUSART_COUNT == 2)
+#define CLOCK_EUSART(ref)                                                                          \
+	(((ref) == EUSART0)   ? cmuClock_EUSART0                                                   \
+	 : ((ref) == EUSART1) ? cmuClock_EUSART1                                                   \
+			      : -1)
+#elif (EUSART_COUNT == 3)
+#define CLOCK_EUSART(ref)                                                                          \
+	(((ref) == EUSART0)   ? cmuClock_EUSART0                                                   \
+	 : ((ref) == EUSART1) ? cmuClock_EUSART1                                                   \
+	 : ((ref) == EUSART2) ? cmuClock_EUSART2                                                   \
+			      : -1)
+#elif (EUSART_COUNT == 4)
+#define CLOCK_EUSART(ref)                                                                          \
+	(((ref) == EUSART0)   ? cmuClock_EUSART0                                                   \
+	 : ((ref) == EUSART1) ? cmuClock_EUSART1                                                   \
+	 : ((ref) == EUSART2) ? cmuClock_EUSART2                                                   \
+	 : ((ref) == EUSART3) ? cmuClock_EUSART3                                                   \
```

Comment:
-1 to these hacks, please, describe clocks properly in DT.

Reply:
It looks like you have a CMU (Clock Management Unit) peripheral, so, a clock controller. I guess you can encode the clock id in a cell as many other vendors do, eg `eusart0: xxx@uuu { ... clocks = <&cmu CMU_CLOCK_EUSART0>; };`

Reply:
Indeed, it is the idea. I am not saying it is a huge work (even if I didn't check the details).

Reply:
Indeed, maybe. @Terezvent are you fine with that?

Reply:
@Terezvent, I suggest hold this PR a bit. Silabs works on the driver for the EFR clock. A first version is going to be submitted soon.

Reply:
I've started looking at implementing clock control properly for EFR32 Series 2. My WIP depends on the new clock_manager API in the latest Simplicity SDK, so it relies on https://github.com/zephyrproject-rtos/zephyr/pull/78723 getting merged first.

I wouldn't want @Terezvent to end up doing work here that becomes wasted, so it comes down to a question whether they want to create a minimal clock control implementation to get this PR through, or wait for the full clock control driver.

Reply:
Hi everyone, thanks for your comments!
@asmellby do you know how long it would take to develop the clock control driver?

Reply:
I have a working implementation/proof of concept at https://github.com/zephyrproject-rtos/zephyr/compare/main...asmellby:zephyr:feature/clock-control-s2. But it relies on the HAL update to use the new sl_clock_manager API, and we're seeing some Bluetooth regressions with the new HAL that need to be bottomed out before we can update it.

Reply:
Hi @asmellby I see that your changes for the HAL update are already working, I had started working on a clock control driver but I wanted to know if you think it's worth it for me to continue working on it or if it's better to wait for your change.

Reply:
@Terezvent I've opened a PR with my driver: https://github.com/zephyrproject-rtos/zephyr/pull/79054

Reply:
> @Terezvent I've opened a PR with my driver: #79054

FYI, this is now merged